relay-server/src/actors/global_config.rs

```diff
@@ -263,6 +268,7 @@ impl Service for GlobalConfigService {
                     () = &mut self.fetch_handle => self.update_global_config(),
                     Some(result) = self.internal_rx.recv() => self.handle_result(result),
                     Some(message) = rx.recv() => self.handle_message(message),
+                    _ = shutdown.notified() => break,
```

Comment:
Putting this here means two things:
1. The service will continue to process queued message even after the shutdown signal has been dispatched
2. New messages querying for the global config during the shutdown sequence will fail as soon as the global config service has processed the signal.

I believe both of these are wrong behavior. Shutdown should stop the refresh cycle, which means it should reset the `fetch_handle` and if necessary ensure no new fetch is scheduled. However, the service needs to continue accepting new messages so that other services can wrap up their work.

This also means the test below asserts the wrong behavior.